services/ui-src/src/utils/reports/reports.ts

```diff
@@ -115,6 +116,12 @@ export const compileValidationJsonFromRoutes = (
     // if drawer form present, add validation to schema
     const drawerFormFields = route.drawerForm?.fields.filter(isFieldElement);
     if (drawerFormFields) addValidationToAccumulator(drawerFormFields);
+    if (route.pageType === "modalOverlay") {
+      const overlayFormFields = (
+        route as ModalOverlayReportPageShape
+      ).overlayForm?.fields.filter(isFieldElement);
+      if (overlayFormFields) addValidationToAccumulator(overlayFormFields);
+    }
```

Comment:
It's a little unfortunate that this can't follow the same pattern as `standardFormFields`, `modalFormFields`, and `drawerFormFields`. I'm guessing that the string check on `pageType` here is saving us from a higher-impact change to the `ReportRoute` type?

Reply:
Basically, yes. Because of the way the types are created, we need to cast `route` to the Modal Overlay page shape. We only want to do this if we know it's a modal overlay page, hence the string check.